docs/api-endpoints.md

```diff
@@ -0,0 +1,301 @@
+  | `idp`                 | 'idir' \| 'bceidbasic' \| 'bceidbusiness'\| 'bceidboth' | path  | The identity provider the users associated with |
+  | `search`              | string                                                  | path  | The search key to use for the search criteria   |
+  | `property` (optional) | 'email' \| 'firstName' \| 'lastName' \| 'guid'          | query | The search property (defaults to 'email')       |
+
+- Responses
+
+<table style="margin-left: 2em;">
+<tr><td>Status Code</td><td>Response</td></tr>
+<tr>
+<td>200</td>
+<td>
+
+```json
+[
+  {
+    "username": "f9468f2708443a0729704d475f1e8bc3@idir",
+    "firstName": "James",
+    "lastName": "Smith",
+    "email": "james.smith@gov.bc.ca",
+    "attributes": {
+      "idir_user_guid": "F9468F2708443A0729704D475F1E8BC3",
+      "idir_username": "jasmith"
+    }
+  }
+]
+```
+
+</td>
+</tr>
+<tr>
+<td>422</td>
+<td>
+
+```json
+{ "success": false, "message": "<error_message>" }
+```
+
+</td></tr>
+</table>
+
+#### Create roles for the integration
+
+```
+POST /api/integrations/{id}/roles
+```
+
+- Parameters
+
+  | Name    | Type        | In   | Description               |
+  | ------- | ----------- | ---- | ------------------------- |
+  | `id`    | number      | path | The id of the integration |
+  | `roles` | `NewRole`[] | body | Array of roles            |
+
+  ```ts
+  interface NewRole {
+    name: string;
+    envs: 'dev' | 'test' | 'prod';
+  }
+  ```
+
+- Responses
+
+<table style="margin-left: 2em;">
+<tr><td>Status Code</td><td>Response</td></tr>
+<tr>
+<td>200</td>
+<td>
+
+```json
+[
+  {
+    "env": "dev",
+    "success": ["admin", "manager"],
+    "duplicate": [],
+    "failure": [],
+    "clientNotFound": false
+  },
+  {
+    "env": "test",
+    "success": ["admin"],
+    "duplicate": ["manager"],
+    "failure": [],
+    "clientNotFound": false
+  },
+  {
+    "env": "prod",
+    "success": [],
+    "duplicate": [],
+    "failure": ["admin", "manager"],
+    "clientNotFound": true
+  }
+]
+```
+
+</td>
+</tr>
+<tr>
+<td>422</td>
+<td>
+
+```json
+{ "success": false, "message": "<error_message>" }
+```
+
+</td></tr>
+</table>
+
+#### Delete a role for the integration
+
+```
+DELETE /api/{environment}/integrations/{id}/roles/{roleName}
+```
+
+- Parameters
+
+  | Name          | Type                      | In   | Description                            |
+  | ------------- | ------------------------- | ---- | -------------------------------------- |
+  | `environment` | 'dev' \| 'test' \| 'prod' | path | The environment to search the users in |
+  | `id`          | number                    | path | The id of the integration              |
+  | `roleName`    | string                    | path | The role name to delete                |
+
+- Responses
+
+<table style="margin-left: 2em;">
+<tr><td>Status Code</td><td>Response</td></tr>
+<tr>
+<td>200</td>
+<td>
+
+```json
+// the deleted role name
```

Comment:
is this supposed to be JSON

Reply:
I moved the comment out of the json block.